snapcraft/internal/lxd/_project.py

```diff
@@ -73,7 +73,11 @@ def _ensure_container(self):
             # Because of https://bugs.launchpad.net/snappy/+bug/1628289
             # Needed to run snapcraft as a snap and build-snaps
             self._container_run(['apt-get', 'install', 'squashfuse', '-y'])
+        if self._remote != 'local':
+            # For remote mounts we will need sshfs.
+            self._container_run(['apt-get', 'install', 'sshfs', '-y'])
         self._inject_snapcraft(new_container=new_container)
+        self._setup_vendoring()
```

Comment:
At least the last few lines here should probably be refactored into something callable from both here and _cleanbuilder, or we have a good opportunity for diversion.